test/util/ilist_test.cpp

```diff
@@ -65,6 +65,46 @@ TestList BuildList(TestNode nodes[], int n) {
   return list;
 }
 
+// Test decrementing begin()
+TEST(IListTest, DecrementingBegin) {
```

Comment:
Thanks for these extra tests.

The wraparound behaviour of the iterators is good to document, but hopefully nobody relies on it.